src/worker/mod.rs

```diff
@@ -433,9 +413,13 @@ impl Worker {
     /// callers can't do anything about.
     async fn workflow_poll(&self) -> Result<Option<ValidPollWFTQResponse>, PollWfError> {
         // We can't say we're shut down if there are outstanding LAs, as they could end up WFT
-        // heartbeating which is a "new" workflow task that we need to accept and process as long
-        // as the LA is outstanding.
-        if *self.wfts_drained.borrow() && self.local_act_mgr.num_outstanding() == 0 {
+        // heartbeating which is a "new" workflow task that we need to accept and process as long as
+        // the LA is outstanding. Similarly, if we already have such tasks (from a WFT completion),
+        // then we must fetch them from the source before we can say workflow polling is shutdown.
+        if *self.shutdown_requested.borrow()
+            && !self.wf_task_source.has_tasks_from_complete()
```

Comment:
could there be any buffered tasks in the poller of the wf task source?

Reply:
Maaaaaybe. It shouldn't happen b/c the pollers should stop fetching new work as soon as we initiate shutdown. Worst case if somehow we were so heavily overloaded we never get to them, they'll just time out.